js/app/articleHTMLRenderer.js

```diff
@@ -179,134 +69,51 @@ var ArticleHTMLRenderer = new Knowledge.Class({
 
-    _get_wrapper_css_files: function () {
-        return ['clipboard.css', 'share-actions.css'].concat(this._custom_css_files);
-    },
-
-    _get_wrapper_js_files: function () {
-        return [
-            'jquery-min.js',
-            'clipboard-manager.js',
-            'crosslink.js',
-            'chunk.js',
-            'share-actions.js',
-        ];
-    },
-
-    _get_crosslink_data: function (model) {
+    _render_wrapper: function (content, model) {
+        let setMap = null;
+        let tagMap = SetMap.get_tag_map();
         let engine = Eknc.Engine.get_default();
-        let links = model.outgoing_links.map((link) => engine.test_link(link));
-        return JSON.stringify(links);
-    },
-
-    _get_chunk_data: function (model) {
-        function get_parent_featured_sets () {
-            return model.tags
-                .filter(tag => !tag.startsWith('Ekn'))
-                .map(tag => SetMap.get_set_for_tag(tag))
-                .filter(set => typeof set !== 'undefined')
-                .filter(set => set.featured);
-        }
-
-        return JSON.stringify({
-            // Need to explicitly expand properties, stringify does not
-            // work on GI-binded objects
-            'ParentFeaturedSets': get_parent_featured_sets().map((set) => ({
-                child_tags: set.child_tags,
-                ekn_id: set.ekn_id,
-                title: set.title,
-                tags: set.tags,
-            })),
-        });
-    },
-
-    _get_metadata: function (model) {
-        let metadata = {
-            id: model.ekn_id,
-            title: model.title,
-            published: model.published,
-            authors: model.authors,
-            license: model.license,
-            source: model.source,
-            source_name: model.source_name,
-            original_uri: model.original_uri,
-            sets: [],
-        };
-
-        // augment sets data to make it useful
-        model.tags.forEach((tag) => {
-            if (tag.startsWith('Ekn'))
-                return;
-            let set = SetMap.get_set_for_tag(tag);
-            if (!set)
-                return;
-            metadata['sets'].push({
-                id: set.ekn_id,
-                title: set.title,
-                featured: set.featured,
-            });
-        });
-
-        return JSON.stringify(metadata);
-    },
 
-    _get_share_actions_markup: function (model) {
-
-        if (!model.original_uri || model.original_uri === '')
-            return '';
-
-        function get_button_markup (network) {
-            let file = Gio.file_new_for_uri(`resource:///com/endlessm/knowledge/data/icons/scalable/apps/${network}-symbolic.svg`);
-            let [success, svg] = file.load_contents(null);
-            return `
-            <a class="share-action"
-               onclick="window.webkit.messageHandlers.share_on_${network}.postMessage(0)">
-               ${svg.toString()}
-            </a>`;
+        if (tagMap) {
```

Comment:
I'm thinking move this into the previous commit and add it as a `get_legacy_boxed_set_info()` method of `setMap.js`. That way it can be more easily unit tested.